traits/trait_set_object.py

```diff
@@ -330,15 +309,14 @@ def difference_update(self, value):
                 An empty set.
 
         """
-        validated_values = self.validate(value)
         removed = self.intersection(value)
 
         if len(removed) > 0:
-            super().difference_update(validated_values)
+            super().difference_update(removed)
```

Comment:
I look at this in 2 ways:
- The `value` is a python `set`
- The `value` is a `TraitSet`


If the `value` passed into this method is going to be a python `set`, then we can assume that those values would not have undergone a transformation when being created, so we need to transform them before computing the `difference_update`.

If the `value` is a `TraitSet`, that means those values were already transformed previously and we do not need to call `validate` anymore, in which case this is right.

How are we looking at `value` here ? Also applies to other `*_update` methods.

Reply:
My understanding is that only values to be added should undergo transformation provided by the validator. If I try to remove values from the set, I don't expect the values to be removed to be first transformed by the validator before being matched with values in the set.